docs/users.md

```diff
@@ -314,6 +315,82 @@ To create a control plane of type `host` run the command:
 kflex create cp4 --type host
 ```
 
+To create a control plane of type `external` with the required options, run the command: 
+
+```shell
+kflex adopt --adopted-context <kubeconfig-context-of-external-cluster> cp5
+```
+
+*Important*: This command generates a secret containing a long-lived token for accessing 
+the external cluster within the namespace associated with the control plane. The secret is automatically 
+removed when the associated control plane is deleted.
+
+### Creating a control plane of type `external` with the API
+
+To create a control plane of type `external` with the API, you need to provide 
+first a **bootstrap secret** containing a bootstrap Kubeconfig for accessing the external cluster.
+The bootstrap Kubeconfig is used by the KubeFlex controllers to generate a long-lived
+token for accessing the external cluster.  The bootstrap kubeconfig is required to have only one context, 
+so given a Kubeconfig for the external cluster `$EXTERNAL_KUBECONFIG` with context for the external
+cluster `$EXTERNAL_CONTEXT` you can generate the `$BOOTSTRAP_KUBECONFIG` with the command:
+
+```shell
+kubectl --kubeconfig=$EXTERNAL_KUBECONFIG config view --minify --flatten \
+--context=$EXTERNAL_CONTEXT > $BOOTSTRAP_KUBECONFIG
+```
+
+If the Kubeconfig for your external cluster uses a loopback address for the server URL, you
+need to follow these [steps](#determining-the-endpoint-for-an-external-cluster-using-loopback-address) 
+to determine the address to use for `cluster.server` in the Kubeconfig and set that value in
+the file referenced by`$BOOTSTRAP_KUBECONFIG` created in the previous step. If the address is the value of `$INTERNAL_ADDRESS` then you can update the bootstrap Kubeconfig as follows:
+
+```shell
+# e.g. INTERNAL_ADDRESS=https://ext1-control-plane:6443
+kubectl --kubeconfig=$BOOTSTRAP_KUBECONFIG config set-cluster $(kubectl --kubeconfig=$BOOTSTRAP_KUBECONFIG config current-context) --server=$INTERNAL_ADDRESS
+```
+
+At this point, you can create the bootstrap secret with the command:
+
+```shell
+CP_NAME=ext1
+kubectl create secret generic ${CP_NAME}-bootstrap --from-file=kubeconfig-incluster=$BOOTSTRAP_KUBECONFIG --namespace kubeflex-system
+```
+where `${CP_NAME}` is the name of the control plane to create.
+
+*Important*: once the KubeFlex controller generates a long-lived token, it removes the bootstrap secret.
+
+Finally, you can create the new control plane of type "external" applying the following yaml:
+
+```shell
+kubectl apply -f - <<EOF
+apiVersion: tenancy.kflex.kubestellar.org/v1alpha1
+kind: ControlPlane
+metadata:
+  name: ${CP_NAME}
+spec:
+  type: external
+  bootstrapSecretRef:
+    inClusterKey: kubeconfig-incluster
+    name: ${CP_NAME}-bootstrap
+    namespace: kubeflex-system
+EOF
+```
+You can verify that the control plane has been created correctly with the command:
+
+```shell
+$ kubectl get cps
+NAME   SYNCED   READY   TYPE       AGE
+ext1   True     True    external   5s
+```
```

Comment:
FYI, GitHub displays this better if the language tag is "console". For example, see how the command and the output are colored differently below.

```console
$ kubectl get cps
NAME   SYNCED   READY   TYPE       AGE
ext1   True     True    external   5s
```

Reply:
ok, updated in all the places where the text shows the command and the output.